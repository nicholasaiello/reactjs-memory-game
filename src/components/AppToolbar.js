import React from 'react';

import Avatar from 'material-ui/Avatar';
import Chip from 'material-ui/Chip';
import RaisedButton from 'material-ui/RaisedButton';
import {Toolbar, ToolbarGroup, ToolbarSeparator, ToolbarTitle} from 'material-ui/Toolbar';

const styles = {
  chip: {
    margin: '0 4px 0 24px',
    color: '#fff',
    textTransform: 'uppercase',
    fontFamily: "'Lato', sans-serif"
  },
  button: {
    marginRight: '0px'
  }
};

const AppToolbar = ({ gameStarted, gameStats = {}, onStartClick, onEndClick }) => {

  let actions;
  if (!gameStarted) {
    actions = (<ToolbarGroup>
      <ToolbarSeparator />
      <RaisedButton 
          label={"Start a Game"}
          style={styles.button} 
          primary={true}
          onClick={onStartClick} />
      </ToolbarGroup>);
  } else {
    actions = (<ToolbarGroup>
      <ToolbarSeparator />
        <Chip
          backgroundColor={'#eefbff'}
          style={styles.chip}>
          <Avatar size={32} color={'#eefbff'} backgroundColor={'#101046'}>
            {gameStats.attempts || 0}
          </Avatar>
          {"Attempts"}
        </Chip>
        <Chip
          backgroundColor={'#eefbff'}
          style={styles.chip}>
          <Avatar size={32} color={'#eefbff'} backgroundColor={'#101046'}>
            {gameStats.matches || 0}
          </Avatar>
          {"Matches"}
        </Chip>
        <ToolbarSeparator />
        <RaisedButton 
          label={"End Game"}
          style={styles.button} 
          primary={false}
          onClick={onEndClick} />
      </ToolbarGroup>);
  }

  return (
    <Toolbar className={"toolbar"}>
        <ToolbarGroup firstChild={true}>
          <ToolbarTitle className={"toolbar-title"} text={"ReactJS Memory Game"} />
        </ToolbarGroup>
        {actions}
      </Toolbar>
  );
};

export default AppToolbar;
