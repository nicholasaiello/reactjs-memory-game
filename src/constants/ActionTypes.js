/**
 * Game actions
 */

// game
export const NEW_GAME = 0;
export const END_GAME = 1;
export const WON_GAME = 2;

// cards
export const DEFAULT_CARD = 3;
export const FLIP_CARD = 4;
export const MATCH_CARD = 5;
export const NO_MATCH_CARD = 6;

// stats
export const UPDATE_STATS = 7;
