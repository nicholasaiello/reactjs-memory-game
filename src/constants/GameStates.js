/**
 * Game States
 */
export const INACTIVE = 0;
export const ACTIVE = 1;
export const ENDING = 2
export const COMPLETED = 3;
export const ENDED = 4;
