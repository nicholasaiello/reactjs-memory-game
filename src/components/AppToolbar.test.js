import React from 'react';
import renderer from 'react-test-renderer';

import * as GameStates from '../constants/GameStates';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppToolbar from './AppToolbar';

describe('AppToolbar', () => {

  it('renders toolbar for inactive game', () => {
    const state = {state: GameStates.INACTIVE, prevState: GameStates.INACTIVE};
    const component = renderer.create(
      <MuiThemeProvider>
        <AppToolbar stats={{}} gameState={state} onStartClick={() => {}} onEndClick={() => {}} />
      </MuiThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders toolbar for active game', () => {
    const stats = {attempts: 0, matches: 0};
    const state = {state: GameStates.ACTIVE, prevState: GameStates.ACTIVE};

    const component = renderer.create(
      <MuiThemeProvider>
        <AppToolbar stats={stats} gameState={state} onStartClick={() => {}} onEndClick={() => {}} />
      </MuiThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
