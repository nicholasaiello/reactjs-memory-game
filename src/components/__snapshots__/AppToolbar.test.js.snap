// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar renders toolbar for active game 1`] = `
<div
  className="toolbar"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": -24,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <span
      className="toolbar-title"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.4)",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 20,
          "lineHeight": "56px",
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingRight": 16,
          "position": "relative",
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      ReactJS Memory Game
    </span>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": undefined,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <span
      className={undefined}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.175)",
          "color": "rgba(0, 0, 0, 0.4)",
          "display": "block",
          "height": 32,
          "lineHeight": "56px",
          "marginLeft": 24,
          "muiPrepared": true,
          "width": 1,
        }
      }
    />
    <div
      className="toolbar-btn"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "margin": "10px 24px",
          "minWidth": 88,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#00bcd4",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Start a Game
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`AppToolbar renders toolbar for inactive game 1`] = `
<div
  className="toolbar"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": -24,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <span
      className="toolbar-title"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.4)",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 20,
          "lineHeight": "56px",
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingRight": 16,
          "position": "relative",
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      ReactJS Memory Game
    </span>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": undefined,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <span
      className={undefined}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.175)",
          "color": "rgba(0, 0, 0, 0.4)",
          "display": "block",
          "height": 32,
          "lineHeight": "56px",
          "marginLeft": 24,
          "muiPrepared": true,
          "width": 1,
        }
      }
    />
    <div
      className="toolbar-btn"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "margin": "10px 24px",
          "minWidth": 88,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#00bcd4",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Start a Game
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
